demo/src/main/java/com/novoda/movies/rateable/RateableMovieViewHolder.java

```diff
@@ -0,0 +1,120 @@
+package com.novoda.movies.rateable;
+
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.RatingBar;
+import android.widget.TextView;
+
+import com.novoda.accessibility.AccessibilityServices;
+import com.novoda.accessibility.Action;
+import com.novoda.accessibility.Actions;
+import com.novoda.accessibility.ActionsAccessibilityDelegate;
+import com.novoda.accessibility.ActionsAlertDialogCreator;
+import com.novoda.movies.R;
+
+import java.util.Arrays;
+
+class RateableMovieViewHolder extends RecyclerView.ViewHolder {
+
+    static RateableMovieViewHolder inflate(ViewGroup parent) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.item_view_rateable_movie, parent, false);
+        return new RateableMovieViewHolder(view);
+    }
+
+    private final ImageView posterImageView;
+    private final TextView titleTextView;
+    private final ImageView likeImageView;
+    private final RatingBar ratingBar;
+    private final AccessibilityServices a11yServices;
+
+    RateableMovieViewHolder(View itemView) {
+        super(itemView);
+        posterImageView = itemView.findViewById(R.id.item_rateable_image_poster);
+        titleTextView = itemView.findViewById(R.id.item_rateable_text_title);
+        likeImageView = itemView.findViewById(R.id.item_rateable_image_like);
+        ratingBar = itemView.findViewById(R.id.item_rateable_rating);
+        a11yServices = AccessibilityServices.newInstance(itemView.getContext());
+    }
+
+    void bind(RateableMovieViewModel viewModel) {
+        posterImageView.setImageResource(viewModel.poster);
+        titleTextView.setText(viewModel.title);
+        likeImageView.setImageResource(viewModel.liked ? R.drawable.ic_favorite_24dp : R.drawable.ic_favorite_border_24dp);
+        ratingBar.setRating(viewModel.rating);
+        Actions actions = collateActionsFor(viewModel);
+        ActionsAccessibilityDelegate a11yDelegate = new ActionsAccessibilityDelegate(itemView.getResources(), actions);
+        ViewCompat.setAccessibilityDelegate(itemView, a11yDelegate);
+        itemView.setContentDescription(viewModel.title + ", rating " + viewModel.rating + ", liked: " + viewModel.liked);
```

Comment:
the four lines above here may be new to some people. The `Actions` class is from accessibilitools - it's used to define each action that's available on a given view. This `Action` is comprised of:

- an ID (must be a unique id for the view, so we use resource ids to ensure uniqueness)
- a string resource - the label for the action
- a runnable to do the action (which in our case is a callback, available on the viewmodel)

here's the one for the select (click itemview) action:

```java
private Action selectActionFor(RateableMovieViewModel viewModel) {
    return new Action(
            R.id.action_rateable_movie_click_select,
            R.string.action_rateable_movie_click_select,
            () -> viewModel.actions.onSelectMovie()
    );
}
```

accessibilitools wants these because it uses the Actions class in two places:

- the ActionsAccessibilityDelegate, which exposes these actions to the system, and to any accessibility service that is running
- the ActionsAlertDialogCreator, which creates an alert dialog containing these actions in a list

Setting this (particular) accessibility delegate and setting a content description will not change the behaviour for direct touch users - they just expose more information to any running accessibility services. so we can safely do it with the other stuff above.